test/libsolidity/SyntaxTest.cpp

```diff
@@ -188,6 +188,9 @@ int SyntaxTest::registerTests(
 	}
 	else
 	{
+		static std::vector<std::string> filenames;
```

Comment:
This assumes that the data area of the string does not change when the object itself is moved. This is probably the case, but perhaps it would be better to use unique_ptrs here.